site/archive/versions.ts

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from "react"
+import * as R from "remeda"
+import { useVersionsQuery } from "./queries.js"
+
+export function useArchiveVersions(versionsFileUrl?: string) {
+    const { data } = useVersionsQuery(versionsFileUrl)
+
+    const versions = useMemo(() => {
+        const versionsObj = data
+        if (
+            !R.isPlainObject(versionsObj) ||
+            !("versions" in versionsObj) ||
+            !Array.isArray(versionsObj.versions)
+        )
+            return []
+
+        return R.sortBy(versionsObj.versions, [
+            (item) => item.archivalDate,
+            "desc",
+        ])
```

Comment:
I guess we can just assume that, by contract, they are sorted _ascending_, and then rely on that info going forward?